vmdb/app/models/host.rb

```diff
@@ -788,18 +788,18 @@ def refreshable_status
 
   def scannable_status
     s = self.refreshable_status
-    if s[:show] == false && s[:enabled] == false
-      if self.authentication_valid?(:ipmi) || !self.ipmi_address.blank?
-        if self.authentication_invalid?(:ipmi)
-          s.merge!(:show => true, :enabled => false, :message => "Provide credentials for IPMI")
-        elsif self.ipmi_address.blank?
-          s.merge!(:show => true, :enabled => false, :message => "Provide an IPMI Address")
-        else
-          s.merge!(:show => true, :enabled => true, :message => "")
-        end
-      end
+    return s if s[:show] || s[:enabled]
+
+    # TODO: Write tests for this
+    if self.has_credentials?(:ipmi) && !self.ipmi_address.blank?
+      s.merge!(:show => true, :enabled => true, :message => "")
+    elsif self.ipmi_address.blank?
+      s.merge!(:show => true, :enabled => false, :message => "Provide an IPMI Address")
+    elsif self.missing_credentials?(:ipmi)
+      s.merge!(:show => true, :enabled => false, :message => "Provide credentials for IPMI")
     end
-    return s
+
+    s
```

Comment:
@Fryguy @blomquisg does this look a bit better?  I still have to write tests for this but wanted to get started on this.